Extensions/retags.js

```diff
@@ -19,111 +19,168 @@ XKit.extensions.retags = {
 			this.blog_name = $('body').data('new-root').split('/').pop();
 		} catch(e) {}
 		this.add_toggle();
-		this.observer.observe($('body')[0],{childList:true,subtree:true});
+		this.observer.observe($('body')[0],{
+			childList:true,
+			subtree:true
+		});
 		this.tag(this.selectors);
 	},
 
 	observer: new MutationObserver(function(ms){
-		var inner, $el;
+		var $baseMutatedElements;
+		var notesProcessed = false;
 		ms.forEach(function(m){
-			$el = $(m.addedNodes);
-			inner = $el.find('.note:not(.ui_post_badge)');
-			if (inner.length > 0) {
-				XKit.extensions.retags.tag(inner.filter(XKit.extensions.retags.selectors));
+			$baseMutatedElements = $(m.addedNodes);
+			var nonPopoverNotes = $baseMutatedElements.find('.note:not(.ui_post_badge)');
+			if ($baseMutatedElements.hasClass('note-list')) {
+				//remove existing retags items (if tabbing between popover views)
+				$baseMutatedElements.find('.retags-wrapper').remove();
+				var elements = $baseMutatedElements.find('.post-activity-note');
+				notesProcessed = true;
+				XKit.extensions.retags.tag_popover(elements.filter(XKit.extensions.retags.selectors));
+			} else if ($baseMutatedElements.hasClass('post-activity-note') && !notesProcessed) {
+				XKit.extensions.retags.tag_popover($baseMutatedElements.filter(XKit.extensions.retags.selectors));
+			} else if (nonPopoverNotes.length > 0) {
+				XKit.extensions.retags.tag(nonPopoverNotes.filter(XKit.extensions.retags.selectors));
 			} else {
-				XKit.extensions.retags.tag($el.filter(XKit.extensions.retags.selectors));
+				XKit.extensions.retags.tag($baseMutatedElements.filter(XKit.extensions.retags.selectors));
 			}
 		});
 	}),
 
-	add_toggle: function(){
-		var toggle = 'toggle_' + this.blog_name;
-		if (XKit.browser().mobile) {
-			this.html_toggle.appendTo('.primary-nav');
-			XKit.tools.add_css('label.retags .binary_switch_label {font-size:15px; color:white; padding-bottom:15px; }','retags_mobile_label');
-		} else {
-			this.html_toggle.appendTo('.ui_notes_switcher .part-toggle');
-		}
-		$('#retags-toggle').change(function(){
-			if ($(this).prop('checked')) {
-				XKit.storage.set('retags', toggle, 'true');
-				XKit.extensions.retags.css_toggle.appendTo('head');
-				if (XKit.browser().mobile) {
-					XKit.extensions.retags.mobile_toggle.appendTo('head');
+	tag_popover: function(elements){
+		$(elements).each(function(){
+			var $element = $(this), retagClass, $target, url;
+			if ($element.find('div.retags').length) {
+				return false;
+			}
+			if ($element.hasClass('is_reply') || $element.hasClass('is_answer') || $element.hasClass('type_6')) {
```

Comment:
type_6 was referenced in a previous version of the code before I even got here; I'm not sure what it refers to but there seemed no harm in leaving it as a fallback. I like this rephrase, thanks.